src/rewrite.rs

```diff
@@ -147,6 +148,23 @@ where
             .collect()
     }
 
+    /// For patterns, return the ast directly as a reference
+    fn get_pattern_ast(&self) -> Option<&PatternAst<L>> {
+        None
+    }
+
+    /// Return the pattern corresponding to the searcher.
+    ///
+    /// This pattern is commonly used for proof-production mode.
+    fn get_ast(
```

Comment:
I'm a little bit unhappy with the API right now- often get_ast and apply_one have similar implementations. Not sure how to combine them nicely (see lambda.rs for an example)